### PR TITLE
Remix Navigation Bars

### PR DESCRIPTION
**Pull request by bkrmendy**
image

## Description
This PR adds the navigation bar/label for RemixScene, and "the" navigation bar in the canvas toolbar

### Out of scope
The design of the nav bars, to be applied on a follow-up PR

### CODE REVIEW

File: editor/src/components/editor/canvas-toolbar.tsx [hunk 1]
```diff
@@ -489,8 +490,31 @@ export const CanvasToolbar = React.memo(() => {
                 onClick={selectInsertMenuPane}
               />
             </Tooltip>
+            {/* Live Mode */}
```
Comment (Rheeseyb) [inline editor/src/components/editor/canvas-toolbar.tsx:493]: ?

File: editor/src/components/editor/remix-navigation-bar.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+export const RemixNavigationBarButtonTestId = (button: RemixSceneLabelButtonType): string =>
+  `remix-navigation-bar-button-${button}`
+
+export const RemixNavigationBar = React.memo(() => {
+  const [navigationControls] = useAtom(RemixNavigationAtom)
+  const [activeRemixScene] = useAtom(ActiveRemixSceneAtom)
+
+  const isLiveMode = useEditorState(
+    Substores.restOfEditor,
+    (_) => _.editor.mode.type === 'live',
```
Comment (Rheeseyb) [inline editor/src/components/editor/remix-navigation-bar.tsx:27]: Please don't use the underscore here - we typically use an underscore prefix to indicate a param that is deliberately unused

File: editor/src/components/editor/canvas-toolbar.tsx (hunk 1 above)
Comment (bkrmendy) [reply editor/src/components/editor/canvas-toolbar.tsx]: This is consistent with the rest of the code in the file, the separate sections are broken up by comments such as `{/* Insert Mode */}`, and `{/* Edit Mode submenus */}`. https://github.com/concrete-utopia/utopia/pull/4195/commits/e1ba2ea571479ab8cf8d16e49eff10f030f8b708 moves the comment to the right place though

File: editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+  )
+  const baseFontSize = 9
+  const scaledFontSize = baseFontSize / scale
+  const scaledLineHeight = 17 / scale
+  const paddingY = scaledFontSize / 2
+  const offsetY = scaledFontSize
+  const offsetX = scaledFontSize
+  const borderRadius = 3 / scale
+
+  const storeRef = useRefEditorState((store) => {
```
Comment (gbalint) [inline editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx:121]: Is there a reason why this is not just an editorModeRef?

File: editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+              fontSize: 14 / scale,
+            }}
+          >
+            {label}
+          </div>
+        </FlexRow>
+      </CanvasOffsetWrapper>
+    )
+  } else {
+    return null
```
Comment (gbalint) [inline editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx:281]: Minor stylistic nitpick (feel free to ignore it), but I think a "happy path to the left" early return for the null case is more readable than this else at the end.

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx (hunk 3 above)
Comment (bkrmendy) [reply editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx]: It was the same in `SceneLabel`, renamed it in https://github.com/concrete-utopia/utopia/pull/4195/commits/d320a9a66381c35564cbdacef4baf60a98de2ae9

File: editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx (hunk 4 above)
Comment (bkrmendy) [reply editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx]: refactored it in https://github.com/concrete-utopia/utopia/pull/4195/commits/82830b7462b903d14ae260bc804d14f3ae58515d

**Review by gbalint — APPROVED** [review]

**Pull request merged**